rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/DefaultInfrastructureManager.java

```diff
@@ -138,14 +139,18 @@ public String getDescription() {
      * @return true there is another node in the node's JVM handled by the
      *         nodeSource, false otherwise.
      */
-    public boolean isThereNodesInSameJVM(Node node) {
-        VMID nodeID = node.getVMInformation().getVMID();
-        String nodeToTestUrl = node.getNodeInformation().getURL();
-        Collection<Node> nodesList = nodeSource.getAliveNodes();
-        for (Node n : nodesList) {
-            if (!n.getNodeInformation().getURL().equals(nodeToTestUrl) &&
-                n.getVMInformation().getVMID().equals(nodeID)) {
-                return true;
+    public boolean isThereNodesInSameJVM(RMNode node) {
+        if (node.getNode() != null) {
+            VMID nodeID = node.getNode().getVMInformation().getVMID();
+            String nodeToTestUrl = node.getNodeURL();
+            Collection<RMNode> nodesList = nodeSource.getAliveNodes();
+            for (RMNode otherNode : nodesList) {
+                if (otherNode.getNode() != null) {
+                    if (!otherNode.getNodeURL().equals(nodeToTestUrl) &&
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)